toolchain/parse/testdata/class/fail_introducer.carbon

```diff
@@ -0,0 +1,29 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+
+// CHECK:STDERR: fail_introducer.carbon:[[@LINE+3]]:1: ERROR: Unrecognized declaration introducer.
```

Comment:
The declaration introducer here seems like it should be recognized. I'd expect something more like "`abstract` is only allowed for `class` declarations", "`abstract` may not be combined with `base`", etc. Or at least, "Incorrect use of `abstract` keyword for declaration."

Reply:
I've added a TODO. I think how we address this will depend on how we address the other question of how we parse modifier sequences.